src/lib.rs

```diff
@@ -338,75 +291,95 @@ pub enum Complexity {
-            Self::Cognitive => write!(f, "cognitive"),
-        }
-    }
+/// Thresholds.
+#[derive(Debug, Clone, Copy, Serialize)]
+pub struct Thresholds {
+    wcc: f64,
+    cyclomatic_complexity: f64,
+    cognitive_complexity: f64,
 }
 
-impl FromStr for Complexity {
-    type Err = std::io::Error;
-
-    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
-        match s {
-            "cyclomatic" => Ok(Self::Cyclomatic),
-            "cognitive" => Ok(Self::Cognitive),
-            _ => Err(std::io::Error::new(
-                ErrorKind::Other,
-                format!("Unknown complexity metric: {s}"),
-            )),
-        }
+impl fmt::Display for Thresholds {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "{},{},{}",
+            self.wcc, self.cyclomatic_complexity, self.cognitive_complexity
+        )
     }
 }
 
-impl Complexity {
-    /// All complexity formats.
-    pub const fn all() -> &'static [&'static str] {
-        &["cyclomatic", "cognitive"]
-    }
-
-    /// Default complexity format.
-    pub const fn default_value() -> &'static str {
-        "cyclomatic"
+impl Default for Thresholds {
+    fn default() -> Self {
+        Self {
+            wcc: 60.0,
+            cyclomatic_complexity: 10.0,
+            cognitive_complexity: 10.0,
+        }
     }
 }
 
-/// Thresholds
-#[derive(Debug, Clone, Default)]
-pub struct Thresholds(Vec<f64>);
-
 impl FromStr for Thresholds {
     type Err = std::io::Error;
 
     fn from_str(thresholds: &str) -> std::result::Result<Self, Self::Err> {
-        let parsed_thresholds: std::result::Result<Vec<f64>, _> = thresholds
-            .split(',')
-            .map(|t| t.trim().parse::<f64>())
-            .collect();
-
-        match parsed_thresholds {
-            Ok(values) => Ok(Thresholds(values)),
-            Err(_) => Err(std::io::Error::new(
-                ErrorKind::Other,
-                format!("{thresholds:?} format is invalid."),
-            )),
-        }
+        let mut iter = thresholds.split(',').filter_map(|s| s.parse::<f64>().ok());
+
+        Ok(Self {
+            wcc: iter.next().ok_or(std::io::Error::new(
+                ErrorKind::InvalidInput,
+                format!("Missing or invalid wcc in thresholds: {}", thresholds),
+            ))?,
+            cyclomatic_complexity: iter.next().ok_or(std::io::Error::new(
+                ErrorKind::InvalidInput,
+                format!(
+                    "Missing or invalid cyclomatic complexity in thresholds: {}",
+                    thresholds
+                ),
+            ))?,
+            cognitive_complexity: iter.next().ok_or(std::io::Error::new(
+                ErrorKind::InvalidInput,
+                format!(
+                    "Missing or invalid cognitive complexity in thresholds: {}",
+                    thresholds
+                ),
+            ))?,
+        })
```

Comment:
I think the state has to change because we have to iterate over the first three elements in order to extract `wcc`, `cyclomatic_complexity` and `cognitive_complexity` values. I'm not sure if it's possible to achieve the same result with `peek`